### ISSUE TITLE
[elixir-omg] Watcher fails with unchallenged_exit

### ISSUE THREAD

**Issue by jbunce** [omisego/elixir-omg#347]
Tests fail: https://circleci.com/gh/omisego/elixir-omg/272

Output from deployed environment:

```
2019-01-07 05:20:16,105 INFO    :Starting Launcher
2019-01-07 05:20:16,120 INFO    :Service type to launch is Elixir Watcher
2019-01-07 05:20:16,120 INFO    :Starting launch process for build 67fe9a321141b277d509361700edcf99a246bff8
2019-01-07 05:20:16,129 INFO    :Connected to the Ethereum client
2019-01-07 05:20:16,129 INFO    :Ethereum client is b'{"jsonrpc":"2.0","id":67,"result":"Geth/v1.8.20-stable/linux-amd64/go1.11.2"}\n'
omg_rpc: generated priv/static/swagger.json
omg_watcher: generated priv/static/swagger.json
2019-01-07 05:20:19,129 INFO    :Elixir mix compile successful
2019-01-07 05:20:19,129 INFO    :Using pre-deployed contract on network RINKEBY
2019-01-07 05:20:19,129 INFO    :Writing config_watcher.exs
2019-01-07 05:20:19,161 INFO    :Deleted Watcher LevelDB data
2019-01-07 05:20:22,163 INFO    :Initialised Watcher chain database
2019-01-07 05:20:26,137 INFO    :Watcher Postgres instance initialised
2019-01-07 05:20:26,138 INFO    :Launcher process complete
2019-01-07 05:20:29.514 [info] module=OMG.Watcher.Application function=start_watcher_supervisor/0 ⋅Started application OMG.Watcher.Application⋅
2019-01-07 05:20:29.858 [info] module=OMG.API.EthereumEventListener function=init/1 ⋅Starting EthereumEventListener for depositor⋅
2019-01-07 05:20:29.883 [info] module=OMG.DB function=exit_infos/1 ⋅Reading exits' info, this might take a while. Allowing 60000 ms⋅
2019-01-07 05:20:29.894 [info] module=OMG.API.EthereumEventListener function=init/1 ⋅Starting EthereumEventListener for exit_processor⋅
2019-01-07 05:20:29.896 [info] module=OMG.API.EthereumEventListener function=init/1 ⋅Starting EthereumEventListener for exit_finalizer⋅
2019-01-07 05:20:29.898 [info] module=OMG.API.EthereumEventListener function=init/1 ⋅Starting EthereumEventListener for exit_challenger⋅
2019-01-07 05:20:30.014 [info] module=Phoenix.Endpoint.CowboyHandler function=start_link/3 ⋅Running OMG.Watcher.Web.Endpoint with Cowboy using http://0.0.0.0:7434⋅
2019-01-07 05:20:30.029 [info] module=OMG.DB function=utxos/1 ⋅Reading UTXO set, this might take a while. Allowing 600000 ms⋅
2019-01-07 05:20:30.031 [info] module=OMG.API.State function=init/1 ⋅Started State, height: 0, deposit height: 0⋅
2019-01-07 05:20:30.220 [info] module=OMG.Watcher.BlockGetter.Core function=log_downloading_blocks/2 ⋅Child chain seen at block #43000. Downloading blocks [1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000, 11000, 12000, 13000, 14000, 15000, 16000, 17000, 18000, 19000, 20000, 21000, 22000, 23000, 24000, 25000, 26000, 27000, 28000, 29000, 30000, 31000]⋅
2019-01-07 05:20:30.265 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #1000 DF42265CFCB6ECC9... with 1 txs⋅
2019-01-07 05:20:30.265 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #2000 8BF8E6655129953F... with 1 txs⋅
2019-01-07 05:20:30.266 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #14000 8BA1E699CC91F808... with 6 txs⋅
2019-01-07 05:20:30.267 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #9000 D327F79BC066B820... with 15 txs⋅
2019-01-07 05:20:30.267 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #15000 7C6417D1D45AA00C... with 1 txs⋅
2019-01-07 05:20:30.268 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #5000 6583443CFD3B2D76... with 8 txs⋅
2019-01-07 05:20:30.268 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #4000 72EC5A216BD44C22... with 1 txs⋅
2019-01-07 05:20:30.268 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #12000 3EF9162217A3FDB5... with 4 txs⋅
2019-01-07 05:20:30.268 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #6000 C5C8893F26054CE2... with 1 txs⋅
2019-01-07 05:20:30.268 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #11000 B58DABD92E6F2D0A... with 1 txs⋅
2019-01-07 05:20:30.268 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #13000 259A1A06F6DFE798... with 1 txs⋅
2019-01-07 05:20:30.268 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #3000 521782ECC3DF9134... with 1 txs⋅
2019-01-07 05:20:30.268 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #7000 E97DB557085CB97A... with 7 txs⋅
2019-01-07 05:20:30.268 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #8000 A87578F5E28AE308... with 3 txs⋅
2019-01-07 05:20:30.268 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #10000 E432AF1978DC4676... with 1 txs⋅
2019-01-07 05:20:30.268 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #16000 386033F1A748DD81... with 1 txs⋅
2019-01-07 05:20:30.270 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #18000 1657DC22CF3A4C0E... with 4 txs⋅
2019-01-07 05:20:30.270 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #19000 4B6F1610300317FF... with 1 txs⋅
2019-01-07 05:20:30.271 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #20000 5AAA44DEF02F4F78... with 1 txs⋅
2019-01-07 05:20:30.275 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #22000 F25536AFB66B72C7... with 1 txs⋅
2019-01-07 05:20:30.276 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #21000 DFCF897780ABBFC1... with 10 txs⋅
2019-01-07 05:20:30.276 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #23000 AD136BD2EC628D9B... with 1 txs⋅
2019-01-07 05:20:30.281 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #24000 BE62EA890FC54B98... with 2 txs⋅
2019-01-07 05:20:30.282 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #25000 8DB77B9BF45AD948... with 1 txs⋅
2019-01-07 05:20:30.287 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #29000 EBC0F76A1B6E81FE... with 10 txs⋅
2019-01-07 05:20:30.288 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #26000 9057D0418C12EF53... with 4 txs⋅
2019-01-07 05:20:30.291 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #28000 C0D0EDDEC018890C... with 38 txs⋅
2019-01-07 05:20:30.297 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #17000 9700424909057951... with 1 txs⋅
2019-01-07 05:20:30.302 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #27000 002E4423D9B90F21... with 158 txs⋅
2019-01-07 05:20:30.410 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #30000 22F288B7AD90A9A3... with 759 txs⋅
2019-01-07 05:20:30.467 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #31000 60B267339C4EE52E... with 1020 txs⋅
2019-01-07 05:20:30.476 [info] module=OMG.Watcher.BlockGetter.Core function=log_downloading_blocks/2 ⋅Child chain seen at block #43000. Downloading blocks [32000]⋅
2019-01-07 05:20:30.477 [info] module=OMG.Watcher.BlockGetter.Core function=log_downloading_blocks/2 ⋅Child chain seen at block #43000. Downloading blocks [33000]⋅
2019-01-07 05:20:30.480 [info] module=OMG.Watcher.BlockGetter.Core function=log_downloading_blocks/2 ⋅Child chain seen at block #43000. Downloading blocks [34000]⋅
2019-01-07 05:20:30.483 [info] module=OMG.Watcher.BlockGetter.Core function=log_downloading_blocks/2 ⋅Child chain seen at block #43000. Downloading blocks [35000]⋅
2019-01-07 05:20:30.487 [info] module=OMG.Watcher.BlockGetter.Core function=log_downloading_blocks/2 ⋅Child chain seen at block #43000. Downloading blocks [36000]⋅
2019-01-07 05:20:30.488 [info] module=OMG.Watcher.BlockGetter.Core function=log_downloading_blocks/2 ⋅Child chain seen at block #43000. Downloading blocks [37000]⋅
2019-01-07 05:20:30.492 [info] module=OMG.Watcher.BlockGetter.Core function=log_downloading_blocks/2 ⋅Child chain seen at block #43000. Downloading blocks [38000]⋅
2019-01-07 05:20:30.494 [info] module=OMG.Watcher.BlockGetter.Core function=log_downloading_blocks/2 ⋅Child chain seen at block #43000. Downloading blocks [39000]⋅
2019-01-07 05:20:30.499 [info] module=OMG.Watcher.BlockGetter.Core function=log_downloading_blocks/2 ⋅Child chain seen at block #43000. Downloading blocks [40000]⋅
2019-01-07 05:20:30.502 [info] module=OMG.Watcher.BlockGetter.Core function=log_downloading_blocks/2 ⋅Child chain seen at block #43000. Downloading blocks [41000]⋅
2019-01-07 05:20:30.505 [info] module=OMG.Watcher.BlockGetter.Core function=log_downloading_blocks/2 ⋅Child chain seen at block #43000. Downloading blocks [42000]⋅
2019-01-07 05:20:30.515 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #41000 FDC5E5FD081232E9... with 1 txs⋅
2019-01-07 05:20:30.520 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #42000 1EB45065B380B18C... with 1 txs⋅
2019-01-07 05:20:30.576 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #40000 087C34F40B8AEE44... with 660 txs⋅
2019-01-07 05:20:30.576 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #33000 3DCCFC14576D1FDE... with 1003 txs⋅
2019-01-07 05:20:30.578 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #32000 F333E62BF035058A... with 876 txs⋅
2019-01-07 05:20:30.599 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #38000 05F7F569F751354F... with 926 txs⋅
2019-01-07 05:20:30.624 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #35000 236006C0C62543B2... with 908 txs⋅
2019-01-07 05:20:30.624 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #39000 6E554087C96CEEA7... with 960 txs⋅
2019-01-07 05:20:30.642 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #34000 6B8D2B788B5737E6... with 960 txs⋅
2019-01-07 05:20:30.646 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #37000 F71234F1543860C4... with 909 txs⋅
2019-01-07 05:20:30.650 [info] module=OMG.Watcher.BlockGetter.Core function=validate_download_response/5 ⋅Validating block #36000 289C40A4BF72C5EC... with 1009 txs⋅
2019-01-07 05:21:23.793 [info] module=OMG.API.State.Core function=deposit/2 ⋅Recognized deposits [%{amount: 10, blknum: 1, currency: <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>, eth_height: 3550612, owner: <<75, 107, 231, 155, 226, 180, 118, 223, 207, 172, 170, 104, 148, 137, 210, 32, 201, 209, 236, 66>>}]⋅
2019-01-07 05:21:23.848 [error] module=OMG.Watcher.BlockGetter function=handle_cast/2 ⋅Stopping OMG.Watcher.BlockGetter because of :unchallenged_exit⋅
```

**Comment by pdobacz** [comment]
seems like it got entered in 060123a60f6c3739f9b6b391b3f58e7e9a940b21.

To reproduce, do a fresh sync to the staging env. Doesn't seem to bite on 41558f2aa6755aacd56a18c4825a47d4c58b5b9f

**Comment by pdobacz** [comment]
worth noting that the integration tests managed to pass on https://jenkins.omisego.io/job/omisego/job/elixir-omg/job/master/ but that's probably a false pass related to things happening more slowly on Jenkins

**Comment by pdobacz** [comment]
this issue per se is fixed in #351. Still, the watcher on staging will stop on `:unchallenged_exit` but at least at the right moment. Closing
